flink-connectors/flink-connector-jdbc/src/main/java/org/apache/flink/connector/jdbc/catalog/MySQLCatalog.java

```diff
@@ -0,0 +1,499 @@
+            // PK_NAME maybe null according to the javadoc, generate an unique name in that case
+            pkName = pkName == null ? "pk_" + String.join("_", pkFields) : pkName;
+            return Optional.of(UniqueConstraint.primaryKey(pkName, pkFields));
+        }
+        return Optional.empty();
+    }
+
+    @Override
+    public CatalogBaseTable getTable(ObjectPath tablePath)
+            throws TableNotExistException, CatalogException {
+        if (!tableExists(tablePath)) {
+            throw new TableNotExistException(getName(), tablePath);
+        }
+        try (Connection conn =
+                        DriverManager.getConnection(
+                                baseUrl + tablePath.getDatabaseName(), username, pwd);
+                PreparedStatement ps =
+                        conn.prepareStatement(
+                                String.format("SELECT * FROM %s;", tablePath.getObjectName()))) {
+            DatabaseMetaData metaData = conn.getMetaData();
+            Optional<UniqueConstraint> primaryKey = getPrimaryKey(metaData, null, tablePath);
+            ResultSetMetaData resultSetMetaData = ps.getMetaData();
+
+            String[] columnsClassnames = new String[resultSetMetaData.getColumnCount()];
+            DataType[] types = new DataType[resultSetMetaData.getColumnCount()];
+            for (int i = 1; i <= resultSetMetaData.getColumnCount(); i++) {
+                columnsClassnames[i - 1] = resultSetMetaData.getColumnName(i);
+                types[i - 1] = fromJDBCType(tablePath, resultSetMetaData, i);
+                if (resultSetMetaData.isNullable(i) == ResultSetMetaData.columnNoNulls) {
+                    types[i - 1] = types[i - 1].notNull();
+                }
+            }
+            TableSchema.Builder tableBuilder =
+                    new TableSchema.Builder().fields(columnsClassnames, types);
+            primaryKey.ifPresent(
+                    pk ->
+                            tableBuilder.primaryKey(
+                                    pk.getName(), pk.getColumns().toArray(new String[0])));
+            TableSchema tableSchema = tableBuilder.build();
+            Map<String, String> props = new HashMap<>();
+            props.put(CONNECTOR.key(), IDENTIFIER);
+            props.put(URL.key(), baseUrl + tablePath.getDatabaseName());
+            props.put(TABLE_NAME.key(), tablePath.getObjectName());
+            props.put(USERNAME.key(), username);
+            props.put(PASSWORD.key(), pwd);
+            ps.close();
+            return new CatalogTableImpl(tableSchema, props, "");
+        } catch (Exception e) {
+            throw new CatalogException(
+                    String.format("Failed getting table %s", tablePath.getFullName()), e);
+        }
+    }
+
+    @Override
+    public boolean tableExists(ObjectPath tablePath) throws CatalogException {
+        try {
+            return listTables(tablePath.getDatabaseName()).contains(tablePath.getObjectName());
```

Comment:
This will look up all tables to test existence. We can implement this better by querying directly whether this table exists.